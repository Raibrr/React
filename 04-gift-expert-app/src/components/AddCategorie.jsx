import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategorie = ({onNewCategory}) => {
  
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({target}) => {
      setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) return;
    //setCategories((prevCategories) => [inputValue, ...prevCategories])
    setInputValue('')
    onNewCategory(inputValue.trim())
  }

  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
      <input
        value={inputValue}
        type="text"
        placeholder="Buscar Gift"
        onChange={onInputChange}
      />
    </form>
    )
}

AddCategorie.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}