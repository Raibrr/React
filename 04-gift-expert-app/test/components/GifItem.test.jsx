import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test en <GifItem /> ', () => {

    const url = "test-url"
    const title = "Titulo card"

    test('Debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem url={url} title={title}/>)
        expect(container).toMatchSnapshot()
     })
 })