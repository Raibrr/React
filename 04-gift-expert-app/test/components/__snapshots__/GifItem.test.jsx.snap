// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem />  Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo card"
      src="test-url"
    />
    <p>
      Titulo card
    </p>
  </div>
</div>
`;
